### DISCUSSION TITLE
Google 3D tiles API

### DISCUSSION THREAD

**Discussion by stevenjh** [TerriaJS/terriajs#6785] [Ideas] [answered]
I'm wondering if there's any plans to implement a data type/support for the Google 3D Tiles API?
https://developers.google.com/maps/documentation/tile/3d-tiles

**Comment by na9da** [comment, marked as answer]
Hi @stevenjh,

Short answer:
Looks like you can make Google 3d tiles work with Terria using a catalog definition like below:

```json
{
  "catalog": [
    {
      "name": "Google 3D Tiles",
      "type": "3d-tiles",
      "url": "https://tile.googleapis.com/v1/3dtiles/root.json?key=<API-KEY>"
    }
  ]
}

```
Replace `API-KEY` with your google maps API key (good to restrict it to your domain)

Long answer:
Terria is currently on an older version of CesiumJS. There were a few breaking changes in recent Cesium release which we are yet to integrate into Terria. Here is the ticket [tracking the upgrade](https://github.com/TerriaJS/terriajs/issues/6569). This means, if there were rendering improvements made to Cesium that the tileset is using then it wont be available in the current version of Terria.

**Reply by stevenjh** [reply]
Thanks for the reply!
I gave a similar long answer to my internal request as we're running an even older version of Terria, and I've not got back into the version 8 releases so I'm a bit out of touch with the current status so thanks for elaborating.

**Reply by na9da** [reply]
Yes v7 to v8 was a major change. We have a [v7-to-v8 catalog converter](https://docs.terria.io/guide/contributing/migration-guide/#upgrading-initialization-files-using-the-catalog-converter) to ease some of the pain of upgrading. But it might be more involved if you have code customizations too.

Please let us know in the forums if we can help with something.
